### ISSUE TITLE
B909 false positive when directly returning mutation

### ISSUE THREAD

**Issue by TeamSpen210** [astral-sh/ruff#18399]
### Summary

The B909 rule checks for mutations to loop iterables, but permits cases where `return`/`break` would prevent further iterations. However if the `return` and the mutation are the same statement it flags the value.

Example:
```py
def fail_map(mapping):
    for key in mapping:
        return mapping.pop(key)  # false positive

def success_map(mapping):
    for key in mapping:
        ret = mapping.pop(key)  # passes
        return ret

def fail_list(seq):
    for val in seq:
        return seq.pop(4)  # also false positive
```
Used `ruff check --preview --isolated --select B909 iter_mut_bug.py`.

### Version

ruff 0.11.11 (0397682f1 2025-05-22)

**Comment by ntBre** [comment]
Thanks for the report! Without looking at the code this _seems_ like a pretty straightforward fix since we already handle the second case here.
